chainer/testing/array.py

```diff
@@ -33,15 +33,18 @@ def assert_allclose(x, y, atol=1e-5, rtol=1e-4, verbose=True):
             '  shape: {} {}\n'.format(x.shape, y.shape) +
             '  dtype: {} {}\n'.format(x.dtype, y.dtype))
         if x.shape == y.shape:
-            xx = x if x.ndim != 0 else x.reshape((1,))
-            yy = y if y.ndim != 0 else y.reshape((1,))
+            xx = numpy.atleast_1d(x)
+            yy = numpy.atleast_1d(y)
             err = numpy.abs(xx - yy)
-            i = numpy.unravel_index(numpy.argmax(err), err.shape)
+            tol_err = atol + rtol * err.astype(numpy.float64)
+            i = numpy.unravel_index(
+                numpy.argmax(err.astype(numpy.float64) - tol_err), err.shape)
```

Comment:
Why do you subtract `tol_err`?

Reply:
Ah, `tol_err = atol + rtol * err.astype(numpy.float64)` should be `tol_err = atol + rtol * numpy.abs(y)` I guess.

It's according to this equation in NumPy documentation:
>absolute(a - b) <= (atol + rtol * absolute(b))

https://docs.scipy.org/doc/numpy/reference/generated/numpy.allclose.html

Reply:
>Ah, `tol_err = atol + rtol * err.astype(numpy.float64)` should be `tol_err = atol + rtol * numpy.abs(y)` I guess.

Fixed this.

Reply:
Thanks